src/foam/nanos/boot/NSpecFactory.java

```diff
@@ -98,6 +98,7 @@ void initService(X x, Object ns) {
     }
 
     X  nx = x_ instanceof SubX ? x_ : x_.getX();
+    Object ns = ns_;
```

Comment:
do we need this, can you not use ns_ below instead of ns?

Reply:
ns is used like a pointer to traverse through the delegates and make updates to them if needed. We do not want to use ns_ because want it to point to the "head".